GitCommands/Git/GitRevisionTester.cs

```diff
@@ -15,7 +16,7 @@ public interface IGitRevisionTester
         /// <returns>
         /// True if one of the first selected is parent
         /// </returns>
-        bool AllFirstAreParentsToSelected(IEnumerable<GitRevision> firstSelected, GitRevision selectedRevision);
+        bool AllFirstAreParentsToSelected(IReadOnlyList<ObjectId> firstSelected, GitRevision selectedRevision);
```

Comment:
Why?

Reply:
In some situations (for instance in FormCommit), the GitRevision need to be recreated, just to be converted to ObjectId and strings later.
There are further changes in #7899 and beyond, I have done some conversions previously to.

Reply:
Why do we need to change from `IEnumerable<>` to `IReadOnlyList<>`?
This requires additional allocations

Reply:
That is not needed, the list is enumerated to a list though.

Reply:
changed this

Reply:
Need to revert all `.ToList()` in `AllFirstAreParentsToSelected` calls

Reply:
Done